internal/provider/resource_access_secret.go

```diff
@@ -43,10 +43,59 @@ type accessSecretResourceModel struct {
 	SecretId types.String `tfsdk:"secret_id"`
 	// Applications is the list of applications to which the secret is granted or revoked.
 	Applications types.List `tfsdk:"applications"`
+	// ID is used during terraform import.
+	ID types.String `tfsdk:"id"`
 }
 
+// ImportState reads the secret based on the model name and secret name to be
+// imported into terraform.
 func (s *accessSecretResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
-	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+	// model:name
+	parts := strings.Split(req.ID, ":")
+	if len(parts) != 2 || parts[0] == "" || parts[1] == "" {
+		resp.Diagnostics.AddError(
+			"Unexpected Import Identifier",
+			fmt.Sprintf("Expected import identifier with format: <modelname>:<secretname>. Got: %q", req.ID),
+		)
+		return
+	}
+	modelName := parts[0]
+	secretName := parts[1]
+
+	// Prevent panic if the provider has not been configured.
+	if s.client == nil {
```

Comment:
move me at the beginning of the function, no need to do strings.Split if the client is nil.

Reply:
it's a vice versa thing to me. either could fail. moving